src/ui/common/src/components/pages/workflow/id/index.tsx

```diff
@@ -94,18 +100,24 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
   const [showUpdateMessage, setShowUpdateMessage] = useState<boolean>(false);
   const [updateSucceeded, setUpdateSucceeded] = useState<boolean>(false);
 
-  const currentNode = useSelector(
-    (state: RootState) => state.nodeSelectionReducer.selected
+  const selectedNodeState = useSelector(
+    (state: RootState) =>
+      state.workflowPageReducer.perWorkflowPageStates[workflowId]?.SelectedNode
```

Comment:
what does this perWorkflowPageStates do? Does it manage the sidesheet state?

Reply:
After getting deep into most refactors, I think the only thing it's managing is the selected node (which controls whether to show and which sidesheet to show). It doesn't further manage any page states, which will be controlled by page components or already handled by RTK

Reply:
sounds good